commands.js

```diff
@@ -1782,8 +1782,17 @@ var commands = exports.commands = {
 
 	addplayer: function (target, room, user) {
 		if (!target) return this.parse('/help addplayer');
-
-		return this.parse('/roomplayer ' + target);
+		target = this.splitTarget(target);
+		var targetUser = this.targetUser;
+		if (!targetUser) return this.sendReply("User " + this.targetUsername + " not found.");
+		if (!room.joinbattle) return this.sendReply("You can only do this in battle rooms.");
```

Comment:
Changing it to `joinbattle` would cause a crash everywhere, not just in non-battle rooms, incidentally.